fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -1385,6 +1390,66 @@ public LogicalPlan visitRegularQuerySpecification(RegularQuerySpecificationConte
         });
     }
 
+    private LogicalPlan withQualifyQuery(LogicalPlan input, RegularQuerySpecificationContext ctx) {
+        QualifyClauseContext qualifyClauseContext = ctx.qualifyClause();
+        if (qualifyClauseContext != null) {
+            NamedExpression qualifyExpr = visitNamedExpression(qualifyClauseContext.namedExpression());
+            Expression filter;
+            List<NamedExpression> excepts = new ArrayList<>();
+            if (qualifyExpr instanceof UnboundAlias) {
+                Expression cmp = qualifyExpr.child(0);
+                if (!(cmp instanceof ComparisonPredicate)) {
+                    throw new ParseException("only support compare expr");
+                }
+                cmp = ExpressionRuleExecutor.normalize(cmp);
+                Expression left = cmp.child(0);
+                if (left instanceof WindowExpression) {
+                    WindowExpression windowExpression = (WindowExpression) left;
+                    String funName = windowExpression.getFunction().getExpressionName();
+                    switch (funName.toLowerCase()) {
+                        case "row_number":
+                        case "rank":
+                        case "dense_rank":
+                            break;
+                        default:
+                            throw new ParseException("not support window function : " + funName);
+                    }
+                    excepts.add(new UnboundSlot("_QUALIFY_COLUMN"));
+                }
+                if (left instanceof UnboundSlot) {
+                    left = new UnboundSlot(((UnboundSlot) left).getNameParts());
+                } else {
+                    left = new UnboundSlot("_QUALIFY_COLUMN");
+                }
+                Expression right = cmp.child(1);
+                if (!(right instanceof IntegerLikeLiteral)) {
+                    throw new ParseException("qualify expression only support constant compare");
+                }
+                if (cmp instanceof EqualTo) {
+                    filter = new EqualTo(left, right);
+                } else if (cmp instanceof GreaterThan) {
+                    filter = new GreaterThan(left, right);
+                } else if (cmp instanceof GreaterThanEqual) {
+                    filter = new GreaterThanEqual(left, right);
+                } else if (cmp instanceof LessThan) {
+                    filter = new LessThan(left, right);
+                } else if (cmp instanceof LessThanEqual) {
+                    filter = new LessThanEqual(left, right);
+                } else {
+                    throw new ParseException("not support compare type " + cmp.getClass().getSimpleName());
+                }
+            } else {
+                throw new ParseException("not support column type");
+            }
+            LogicalFilter<LogicalPlan> logicalFilter = new LogicalFilter<>(Sets.newHashSet(filter), input);
```

Comment:
should not convert to filter here, because qualify should only allow when window function exists in itself or in project list